context/src/main/java/io/opentelemetry/context/internal/shaded/AbstractWeakConcurrentMap.java

```diff
@@ -283,7 +283,7 @@ public boolean equals(Object other) {
       if (other instanceof WeakKey<?>) {
         return ((WeakKey<?>) other).get() == get();
       } else {
-        return other.equals(this);
+        return other != null && other.equals(this);
```

Comment:
We excluded these two files from checks since they're vendored code and we'd prefer not to diverge for maintenance reasons. Does that make sense?

Reply:
Ah, ok, this makes sense then.  I will drop this PR per your other comment.